### PR TITLE
Add "star/unstar repo" hotkey

### LINKED ISSUE #1479: Keyboard shortcut for starring a repository
What do you think of a keyboard shortcut to start/unstar the current repository?

- When browsing things you could star multiple things and look them up later.
- When you visit a repo from a link while doing something else, you might want to quickly star it and close the page.

Having a keyboard shortcut would allow you to keep your two hands at the keyboard.

<kbd>s</kbd> is already taken for focusing the search bar.

**It could be <kbd>g</kbd><kbd>s</kbd>**. Thoughts?

What do you think of using it toggle the starred state or should it be just for starring?

### PR DESCRIPTION
**Pull request by adamlacombe**
Resolves #1479

### CODE REVIEW

File: source/features/add-star-repo-hotkey.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {registerShortcut} from './improve-shortcut-help';
+import select from 'select-dom';
+
+export default function () {
+	registerShortcut('repos', 'g s', 'Star and unstar repository');
+	let lastKey;
+	document.addEventListener('keydown', (e) => {
+		if ((lastKey === "g" && e.key === "s") || (e.key === "g" && lastKey === "s")) {
```
Comment (notlmn) [inline source/features/add-star-repo-hotkey.js]: `"` -> `'`

File: source/features/add-star-repo-hotkey.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+			let starButtonSelector = (select('div.js-social-container.starring-container').classList.contains('on')) ? 'form.starred.js-social-form > button' : 'form.unstarred.js-social-form > button';
```
Comment (notlmn) [inline source/features/add-star-repo-hotkey.js]: Why not `const`?

File: source/features/add-star-repo-hotkey.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+			select(starButtonSelector).click();
+		}
+		lastKey = e.key;
+	});
+	document.addEventListener('keyup', () => {
+		lastKey = null;
+	})
```
Comment (notlmn) [inline source/features/add-star-repo-hotkey.js]: Missing `;`

File: source/features/add-star-repo-hotkey.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import {registerShortcut} from './improve-shortcut-help';
+import select from 'select-dom';
```
Comment (notlmn) [inline source/features/add-star-repo-hotkey.js]: This should come before all other imports

File: source/features/add-star-repo-hotkey.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import {registerShortcut} from './improve-shortcut-help';
+
+export default function () {
+	registerShortcut('repos', 'g s', 'Star and unstar repository');
+	let lastKey;
+	document.addEventListener('keydown', e => {
+		if ((lastKey === 'g' && e.key === 's') || (e.key === 'g' && lastKey === 's')) {
+			const starButtonSelector = (select('div.js-social-container.starring-container').classList.contains('on')) ? 'form.starred.js-social-form > button' : 'form.unstarred.js-social-form > button';
```
Comment (bfred-it) [inline source/features/add-star-repo-hotkey.js]: Doesn't matter for this PR, but in the future instead `classList.contains` you could use:
```js
select.exists('.js-social-container.starring-container.on')
```

Or avoid the ternary and just select both buttons with

```js
select('form.starred.js-social-form > button, form.unstarred.js-social-form > button')
```

**Comment by sindresorhus** [comment]
@adamlacombe Friendly ping :)

**Comment by sindresorhus** [comment]
@adamlacombe Can you mention the feature in the readme?

**Comment by adamlacombe** [comment]
@sindresorhus Sorry for the delay - should be all set now.

**Comment by bfred-it** [comment]
Thank you for adding the feature, Adam!

**Pull request merged**